src/docbuilder/rustwide_builder.rs

```diff
@@ -1308,4 +1320,29 @@ mod tests {
             Ok(())
         })
     }
+
+    #[test]
+    #[ignore]
+    fn test_new_builder_at_once() {
+        wrapper(|env| {
+            let mut builder = RustwideBuilder::init(env)?.peroid_init(env)?;
+            assert!(builder.build_local_package(Path::new("tests/crates/build-std"))?);
+            Ok(())
+        })
+    }
+
+    #[test]
+    #[ignore]
+    fn test_new_builder_after_interval() {
+        use std::thread::sleep;
+        use std::time::Duration;
+        wrapper(|env| {
```

Comment:
the test environment has `override_config`, which would make the config matching the sleep below more explicit.

Reply:
This would only test if the build just works, without checking if the workspace was reinitialized. 

Let's finish the refactor above, then we can think of a way to actually test the behaviour.

Reply:
I agree with that make the testcase more rubust in the feature.